source/utilities.cc

```diff
@@ -1351,7 +1355,11 @@ namespace aspect
             }
           else
             {
-              std::ifstream filestream(filename.c_str());
+              std::ifstream filestream;
+              if (filename.find(".gz") == filename.size()-3)
```

Comment:
Maybe easier to read, at least for those of us with magic regex skills :-)
```suggestion
              if (std::regex_match (filename, std::regex("\\.gz$"))
```